tasks/main.yml

```diff
@@ -22,6 +27,17 @@
   with_items:
     - src: prometheus.yml.j2
       dest: "{{ prometheus_configuration_directory }}/prometheus/prometheus.yml"
+  when: result.stat.exists == False
+  notify: Reload prometheus configuration
```

Comment:
I think it is better to use with_first_found. This way you do not have to check the stat of the file and you only need one copy/template task.